packages/core/contracts-ovm/insured-bridge/implementation/BridgePoolFactory.sol

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+// Importing local copies of OVM contracts is a temporary fix until the @eth-optimism/contracts package exports 0.8.x
+// contracts. These contracts are relatively small and should have no problems porting from 0.7.x to 0.8.x, and
+// changing their version is preferable to changing this contract to 0.7.x and defining compatible interfaces for all
+// of the imported DVM contracts below.
+import "./OVM_CrossDomainEnabled.sol";
+import "./BridgePoolFactoryInterface.sol";
+import "../../../contracts/oracle/interfaces/IdentifierWhitelistInterface.sol";
+import "../../../contracts/oracle/interfaces/FinderInterface.sol";
+import "../../../contracts/oracle/implementation/Constants.sol";
+import "../../../contracts/common/interfaces/AddressWhitelistInterface.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @notice Administrative contract deployed on L1 that has an implicit reference to a DepositBox. This contract is
+ * responsible for deploying new BridgePools, which houses passive liquidity and enables relaying of L2 deposits.
+ * @dev The owner of this contract can call permissioned functions on the L2 DepositBox.
+ */
+contract BridgePoolFactory is BridgePoolFactoryInterface, Ownable, OVM_CrossDomainEnabled {
```

Comment:
@chrismaree This contract used to be called `BridgeRouter`. Consider how the `OVM_BridgeDepositBox` will interact with this contract:
- This contract will be the cross-chain "owner" of the `OVM_BridgeDepositBox` and can send admin calls
- This contract deploys new `BridgePool` contracts.
- `function bridgeTokens(address l2Token, uint32 l1Gas)` will need to know which `BridgePool` to slow-withdraw funds to. This is the major breaking change.